include/bluetooth/l2cap.h

```diff
@@ -150,6 +150,22 @@ struct bt_l2cap_chan {
 #endif /* CONFIG_BT_L2CAP_DYNAMIC_CHANNEL */
 };
 
+/** @brief Get @c chan->state.
+ *
+ * Returns #BT_L2CAP_CONNECTED if @kconfig{CONFIG_BT_L2CAP_DYNAMIC_CHANNEL} is
+ * disabled.
+ */
+static inline bt_l2cap_chan_state_t bt_l2cap_chan_get_state(
+	struct bt_l2cap_chan *chan)
+{
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
+	return chan->state;
+#else
+	(void)chan;
```

Comment:
Do you really get some warning if you don't do this? And if you really do need to do it Zephyr has a ARG_UNUSED() macro, IIRC (but again, my preference would be to skip unless you get a warning)